apps/estuary/lib/estuary/data_reader.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Estuary.DataReader do
+  @moduledoc """
+  Implementation of `Pipeline.Reader` for Estuary's edges.
+  """
+
+  import Estuary
+
+  @behaviour Pipeline.Reader
+
+  @topic_reader Application.get_env(:estuary, :topic_reader)
+
+  @impl Pipeline.Reader
+  def init(_opts \\ []) do
+    :ok =
+      reader_args()
+      |> @topic_reader.init()
+  end
+
+  @impl Pipeline.Reader
+  def terminate(_opts \\ []) do
+    :ok
```

Comment:
My understanding at the moment (I may have missed something) is that terminate was implemented because the behavior requires it. That being said, we're never calling terminate and so it wasn't doing anything. We'll go ahead and have it call terminate on the writer just so it's consistent with the init / is setup for if/when we do end up using it..